pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -224,13 +221,19 @@ public BrokerResponse handleRequest(JsonNode request, @Nullable RequesterIdentit
       LOGGER.debug("Enable trace for request {}: {}", requestId, query);
       brokerRequest.setEnableTrace(true);
     }
+
     if (request.has(DEBUG_OPTIONS)) {
-      Map<String, String> debugOptions = Splitter.on(';').omitEmptyStrings().trimResults().withKeyValueSeparator('=')
-          .split(request.get(DEBUG_OPTIONS).asText());
+      Map<String, String> debugOptions = getOptionsFromRequest(request, DEBUG_OPTIONS);
       LOGGER.debug("Debug options are set to: {} for request {}: {}", debugOptions, requestId, query);
       brokerRequest.setDebugOptions(debugOptions);
     }
 
+    if (request.has(QUERY_OPTIONS)) {
```

Comment:
Is this the only place we are reading query options?

Reply:
Yes, I didn't find any other place, for all other options as well